packages/core/content-releases/admin/src/components/CMReleasesInjectionZone.tsx

```diff
@@ -0,0 +1,265 @@
+import * as React from 'react';
+
+import {
+  Box,
+  Button,
+  FieldLabel,
+  Flex,
+  ModalBody,
+  ModalHeader,
+  ModalLayout,
+  SingleSelect,
+  SingleSelectOption,
+  Typography,
+  ModalFooter,
+} from '@strapi/design-system';
+import {
+  CheckPermissions,
+  useAPIErrorHandler,
+  useCMEditViewDataManager,
+  useNotification,
+} from '@strapi/helper-plugin';
+import { Plus } from '@strapi/icons';
+import { isAxiosError } from 'axios';
+import { Formik, Form } from 'formik';
+import { useIntl } from 'react-intl';
+import { useParams } from 'react-router-dom';
+import * as yup from 'yup';
+
+import { CreateReleaseAction } from '../../../shared/contracts/release-actions';
+import { PERMISSIONS } from '../constants';
+import { useCreateReleaseActionMutation, useGetReleasesQuery } from '../services/release';
+
+import { ReleaseActionOptions } from './ReleaseActionOption';
+
+const RELEASE_ACTION_FORM_SCHEMA = yup.object().shape({
+  type: yup.string().oneOf(['publish', 'unpublish']).required(),
+  releaseId: yup.string().required(),
+});
+
+interface FormValues {
+  type: CreateReleaseAction.Request['body']['type'];
+  releaseId: CreateReleaseAction.Request['params']['releaseId'];
+}
+
+const INITIAL_VALUES = {
+  type: 'publish',
+  releaseId: '',
+} satisfies FormValues;
+
+interface AddActionToReleaseModalProps {
+  handleClose: () => void;
+}
+
+const AddActionToReleaseModal = ({ handleClose }: AddActionToReleaseModalProps) => {
+  const { formatMessage } = useIntl();
+  const toggleNotification = useNotification();
+  const { formatAPIError } = useAPIErrorHandler();
+  const params = useParams<{ id?: string }>();
+  const {
+    allLayoutData: { contentType },
+  } = useCMEditViewDataManager();
+  // Get all 'pending' releases
+  const { data } = useGetReleasesQuery({
+    filters: {
+      $and: [
+        {
+          releasedAt: {
+            $null: true,
+          },
+        },
+      ],
+    },
+  });
+  const releases = data?.data;
+  const [createReleaseAction, { isLoading }] = useCreateReleaseActionMutation();
+
+  const handleSubmit = async (values: FormValues) => {
+    /**
+     * contentType uid and entry id are not provided by the form but required to create a Release Action.
+     * Optimistically we expect them to always be provided via params and CMEditViewDataManager.
+     * In the event they are not, we should throw an error.
+     */
+    if (!contentType?.uid || !params.id) {
+      toggleNotification({
+        type: 'warning',
+        message: formatMessage({
+          id: 'content-releases.content-manager.notification.entry-error',
+          defaultMessage: 'Failed to get entry',
+        }),
+      });
+
+      return;
+    }
```

Comment:
ℹ️ I'm not really sure the best way to handle this. We don't get the entry information from the form/user we get it from the current contentType / params. Should we just let the backend throw the error or handle it here before the request is made?

Reply:
I think you can safely ignore this check on front, but make sure the backend throws an error if this happens. We can keep it but you're checking this is present two times

Reply:
> We can keep it but you're checking this is present two times

Where am I checking it is present other than here? On the backend?

I think I will need to make sure they are present here or update the types in rtk to accept undefined values for these properties

Reply:
Aaah, sorry, I didn't get it before... 

Yes, we definitely need to check for this to be defined on backend and also here in front to avoid accept undefined values